### DISCUSSION TITLE
why adding poetry-dynamic-versioning to build-system.requires?

### DISCUSSION THREAD

**Discussion by cavedon** [mtkennerly/poetry-dynamic-versioning#160] [Q&A] [answered]
What is the purpose of adding poetry-dynamic-versioning to `build-system`?
Doing that without running `poetry self add  poetry-dynamic-versioning` does not work.
And removing it from `build-system` does not seem to affect the functionality.

Adding to build-system.requires will cause it to be installed in the virtualenv used to build the python package, but it looks like it needs to be installed in the virtualenv where poetry is installed instead...

**Comment by mtkennerly** [comment, marked as answer]
Hi! That setting makes it possible to install your project from its repository using a non-Poetry tool with PEP-517 support, like Pip.

When you use Poetry to manage a project, it ignores the build-system section and automatically does things the Poetry way. However, other tools, like Pip, rely on the build-system config to know what to do.

If someone clones your repo and runs `pip install .` (rather than `poetry install`), then the build-system will tell Pip that it needs to use poetry-core along with the plugin.
